web/auth/auth_config.go

```diff
@@ -0,0 +1,37 @@
+package auth
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+)
+
+type AuthConfig struct {
+	providers map[string]*oauth2.Config
+}
+
+// AuthConfig creates a new OAuth 2.0 config for every enabled OAuth 2.0 provider.
+// Currently only works for GitHub.
+func NewAuthConfig(baseURL string) (*AuthConfig, error) {
+	providers := make(map[string]*oauth2.Config)
+
+	// Enable GitHub.
+	clientID := os.Getenv("GITHUB_KEY")
+	clientSecret := os.Getenv("GITHUB_SECRET")
+	if clientID == "" || clientSecret == "" {
+		return nil, fmt.Errorf("missing GitHub client variables")
+	}
+	githubConfig := &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Endpoint:     github.Endpoint,
+		RedirectURL:  GetCallbackURL(baseURL, "github"),
+		Scopes:       []string{"repo:invite"},
+	}
+	providers["github"] = githubConfig
+	return &AuthConfig{
+		providers: providers,
+	}, nil
+}
```

Comment:
Referring to #663 and new issue #694.

Suggest to remove the map here and simply add a method here like this:
```go
func NewGitHubConfig(baseURL, clientID, clientSecret string) *oauth2.Config {
	return &oauth2.Config{
		ClientID:     clientID,
		ClientSecret: clientSecret,
		Endpoint:     github.Endpoint,
		RedirectURL:  GetCallbackURL(baseURL, "github"),
		Scopes:       []string{"repo:invite"},
	}
}
```
The above func could be called from `main.go` and the `Config` object can be passed in to the `OAuth2` methods and used without having to examine the `r.URL.Path` to get the provider name and looking up in the map.

Reply:
Maybe this new method should be placed in a new file `github.go` and we can add a corresponding `gitlab.go` version later.